### PR TITLE
add ci for supporting ControlPlaneMachineSet on vsphere

### PR DESCRIPTION
**Pull request by WenXinWei**

### CODE REVIEW

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-tp-f28
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-f28

**Comment by WenXinWei** [comment]
Please help review when you have time @gpei @jianlinliu

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml [hunk 1]
```diff
@@ -2786,6 +2786,16 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb
+- as: vsphere-ipi-zones-tp-f28
+  cron: 33 12 4 * *
+  steps:
+    cluster_profile: vsphere-multizone-2
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-ipi'
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2794]:
```suggestion
      E2E_RUN_TAGS: '@vsphere-ipi'
```

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml [hunk 2]
```diff
@@ -2786,6 +2786,16 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb
+- as: vsphere-ipi-zones-tp-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: Also need a destructive test ci job.

**Comment by gpei** [comment]
@WenXinWei I remember we have some additional checking points on CPMS for vSphere zonal cluster, right? so could we add the other required checkings in a standalone step and add it to the dedicated workflow

**Comment by WenXinWei** [comment]
> I remember we have some additional checking points on CPMS for vSphere zonal cluster, right? so could we add the other required checkings in a standalone step and add it to the dedicated workflow

yes. i will add the step in another pr.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml (hunk 2 above)
Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: ok.

**Comment by gpei** [comment]
> yes. i will add the step in another pr.

OK, I found some Azure examples for you like:
[Step:](https://steps.ci.openshift.org/reference/cucushift-installer-check-azure-cpms#title) cucushift-installer-check-azure-cpms
which being referred in Provision chain
[Chain:](https://steps.ci.openshift.org/chain/cucushift-installer-rehearse-azure-stack-ipi-provision#title) cucushift-installer-rehearse-azure-stack-ipi-provision
>[cucushift-installer-check](https://steps.ci.openshift.org/chain/cucushift-installer-check)	This chain is for installer post->install health check.
>[cucushift-installer-check-azure-cpms](https://steps.ci.openshift.org/reference/cucushift-installer-check-azure-cpms)	>Check that CPMS failureDomain setting as expected on single zone or unsupported zone Available on 4.14+.

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh [hunk 3]
```diff
@@ -100,6 +104,11 @@ if ! hasCPMS; then
     exit 0
 fi
 
+if [[ ${CLUSTER_TYPE} == "vsphere" ]] && [[ ${version} == "4.15" ]] && [[ ${FEATURE_SET} != "TechPreviewNoUpgrade" ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh:107]: I think we can merge the condition to the above `hasCPMS` function like:
```
    vsphere*)
       if [[ "${FEATURE_SET}" == "TechPreviewNoUpgrade" ]]; then
          REQUIRED_OCP_VERSION="4.15"
      else
          REQUIRED_OCP_VERSION="4.99999"
       fi
        ;;
```

Comment (WenXinWei) [reply ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh:107]: looks good. thanks for advice.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: also need to add them in 4.16?

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-tp-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-tp-f28-destructive

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml (hunk 2 above)
Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: it's tp in 4.15, 4.16 should be enabled by default. so i think no need.

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: We can not assume the 4.16 case, if 4.16 implement that, then update the ci jobs accordingly on that time, one on e can remember this story if 4.16 did not implement this on the plan.

Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: ok, done. so if it implement in 4.16 as plan. what i need to do in the future is remove the tag-->FEATURE_SET: TechPreviewNoUpgrade.  right?

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: There is already an existing `vsphere-ipi-zones` ci job, once it is implemented in 4.16, you need to drop the tp job and update `REQUIRED_OCP_VERSION` to 4.16

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh (hunk 3 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh:111]: We do not need this if block any more, right?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml (hunk 2 above)
Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:2789]: ok.

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**